android/src/main/kotlin/com/example/user_location_plugin/UserLocationPlugin.kt

```diff
@@ -0,0 +1,131 @@
+        channel.setMethodCallHandler(this)
+        permissionEventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, "permission_event_channel")
+        permissionEventChannel?.setStreamHandler(permissionStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "getPlatformVersion" -> getPlatformVersion(result)
+            "requestPermission" -> requestPermission()
+            "checkPermission" -> checkPermission()
+            else -> result.notImplemented()
+        }
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = binding.activity
+        binding.addActivityResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        TODO("Not yet implemented")
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onDetachedFromActivity() {
+        TODO("Not yet implemented")
+    }
+
+    private fun getPlatformVersion(result: Result) {
+        result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    }
+
+    private fun requestPermission() {
+        context?.run {
+            activity?.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    500
+                )
+            }
+        }
+    }
+
+    private fun checkPermission(): Boolean {
+        var permitted = false
+        context?.run {
+            permitted = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+                    && ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+            permissionEventSource?.success(
+                if (permitted) resources.getText(R.string.permission_granted) else permissionEventSource?.success(
+                    resources.getText(R.string.permission_denied)
+                )
+            )
+        }
+        return permitted
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>?,
+        grantResults: IntArray?
+    ): Boolean {
+        TODO("Not yet implemented")
```

Comment:
This will cause a runtime exception if for any reason the code reaches this line, lets, make use of the function, and change the TODO for a log instead, or something useful

Reply:
This method is not called automatically. I tried to send the response of the requestPermission() via this but it doesn't work. I can remove it and the interface too. What do you think?